server/src/main/java/io/pravega/schemaregistry/service/SchemaRegistryService.java

```diff
@@ -814,43 +815,71 @@ private SchemaInfo normalizeSchemaBinary(SchemaInfo schemaInfo) {
         ByteBuffer schemaBinary = schemaInfo.getSchemaData();
         boolean isValid = true;
         String invalidityCause = "";
+        String type = schemaInfo.getType();
         try {
             String schemaString;
+            String[] tokens;
             switch (schemaInfo.getSerializationFormat()) {
                 case Protobuf:
-                    String[] tokens = NameUtil.extractNameAndQualifier(schemaInfo.getType());
-                    String qualifier = tokens[1]; // this can be empty string if schema info had no qualifier
-                    String name = tokens[0];
+                    tokens = NameUtil.extractNameAndQualifier(schemaInfo.getType());
+                    String protoPackage; 
+                    String protoName;
                     DescriptorProtos.FileDescriptorSet fileDescriptorSet = DescriptorProtos.FileDescriptorSet.parseFrom(
                             schemaInfo.getSchemaData());
+                    if (Strings.isNullOrEmpty(tokens[0])) {
+                        Exceptions.checkArgument(Strings.isNullOrEmpty(tokens[1]), "Package without name not allowed.", schemaInfo.getType());
+                        String[] packageAndName = getPackageAndNameFromProtobuf(fileDescriptorSet);
+                        protoPackage = packageAndName[0];
+                        protoName = packageAndName[1];
+                    } else {
+                        protoPackage = !Strings.isNullOrEmpty(tokens[1]) ? tokens[1] : getPackageAndNameFromProtobuf(fileDescriptorSet)[0];
+                        protoName = tokens[0];
+                    }
+                    type = NameUtil.qualifiedName(protoPackage, protoName);
 
                     isValid = fileDescriptorSet
                             .getFileList().stream()
                             .anyMatch(x -> {
-                                boolean match;
-                                if (x.getPackage() == null) {
-                                    match = Strings.isNullOrEmpty(qualifier);
-                                } else {
-                                    match = x.getPackage().equals(qualifier);
-                                }
-                                return match && x.getMessageTypeList().stream().anyMatch(y -> y.getName().equals(name));
+                                boolean qualifierCheck = Strings.isNullOrEmpty(protoPackage) || protoPackage.equals(x.getPackage());
```

Comment:
nit: could we rename it so that the intent is more explicit? Something like `isQualifierMissing`, etc.